cincinnati/src/plugins/internal/edge_add_remove.rs

```diff
@@ -304,6 +312,83 @@ impl EdgeAddRemovePlugin {
 
         Ok(())
     }
+
+    fn add_conditional_edges(&self, graph: &mut cincinnati::Graph) -> Fallible<()> {
+        let mut conditional_edges: Vec<ConditionalEdge> = vec![];
+        graph
+            .conditional_edges
+            .iter()
+            .try_for_each(|ce| -> Fallible<()> {
+                let mut conditional_edge = ConditionalEdge {
+                    edge_regex: ce.edge_regex.clone(),
+                    edges: vec![],
+                    risks: ce.risks.clone(),
+                };
+
+                let to_regex_string = ce.edge_regex.to.clone();
+                let from_regex_string = ce.edge_regex.from.clone();
+
+                let to_regex = regex::Regex::new(&to_regex_string)
+                    .context(format!("Parsing {} as Regex", &to_regex_string))?;
+                let from_regex = regex::Regex::new(&from_regex_string)
+                    .context(format!("Parsing {} as Regex", &from_regex_string))?;
+
+                let to = graph.find_by_version_regex(&to_regex);
+                to.iter().try_for_each(|(to, to_version)| -> Fallible<()> {
+                    if from_regex_string == ".*" {
+                        let mut froms: Vec<String> = graph
+                            .previous_releases(to)
+                            .map(|(_, _, release)| release.version().to_string())
+                            .collect();
+                        trace!(
+                            "adding conditional edge by regex for '{}': {:?}",
+                            to_version,
+                            froms
+                        );
+                        froms.iter_mut().for_each(|from| {
+                            conditional_edge
+                                .edges
+                                .append(&mut vec![ConditionalUpdateEdge {
```

Comment:
```suggestion
conditional_edge.edges.push(ConditionalUpdateEdge {
```
We're adding a single `ConditionalUpdateEdge`, so instead of creating a mutable Vector we could use `edges.push`